scripts/west_commands/genboard/ncs_genboard.py

```diff
@@ -37,30 +36,34 @@ def do_add_parser(self, parser_adder):
         )
 
         parser.add_argument(
-            "-o", "--output", required=True, type=Path, help="Output directory"
+            "-s", "--json-schema", action="store_true", help="Provide JSON schema"
         )
-        parser.add_argument("-e", "--vendor", required=True, help="Vendor name")
-        parser.add_argument("-b", "--board", required=True, help="Board name")
+
         parser.add_argument(
-            "-d", "--board-desc", required=True, help="Board description"
+            "-r", "--json-schema-response", type=str, help="JSON schema response"
         )
-        parser.add_argument("-s", "--soc", required=True, help="SoC")
-        parser.add_argument("-v", "--variant", required=True, help="Variant")
 
         return parser
 
     def do_run(self, args, unknown_args):
+        with open(SCHEMA, "r") as f:
+            schema = json.loads(f.read())
+
+        if args.json_schema:
+            print(json.dumps(schema))
+            return
+
         with open(CONFIG, "r") as f:
             config = load(f, Loader=Loader)
 
         # validate input
-        if not VENDOR_RE.match(args.vendor):
-            log.err(f"Invalid vendor name: {args.vendor}")
-            return
+        input = json.loads(args.json_schema_response)
```

Comment:
This throws if command is called without any parameters either. It would be nice to show help instead.

Reply:
fixed